packages/next/server/web/spec-extension/response.ts

```diff
@@ -7,6 +7,25 @@ import { NextCookies } from './cookies'
 const INTERNALS = Symbol('internal response')
 const REDIRECTS = new Set([301, 302, 303, 307, 308])
 
+function handleMiddlewareField(
+  init: MiddlewareResponseInit | undefined,
+  headers: Headers
+) {
+  if (init?.request?.headers) {
+    if (!(init.request.headers instanceof Headers)) {
+      throw new Error('request.headers must be an instance of Headers')
+    }
```

Comment:
It would be useful but I'm not inclined to support it in this PR because:

- It's not obvious what it does: does it *add* headers to existing ones or does it *replace* all existing headers with the object?
- Ambiguity of headers: what if, for example, the user specify `{ Cookie: 'a=b', cookie: 'c=d' }`?

`Headers` nicely prevents these traps. I think *adding* new headers through specifying an object here is convenient but I'd hear feedbacks to be sure 😄

Reply:
I agree we haven't discussed that. In my opinion, it should behave exactly the same, that is, replacing all headers.
Let's sleep on it.